lib/inferno/apps/web/controllers/test_runs/create.rb

```diff
@@ -21,13 +21,13 @@ def call(params)
               return
             end
 
-            # TODO: This test run shouldn't be created until after the inputs
-            # and runnable are validated
-            test_run = repo.create(create_params(params).merge(status: 'queued'))
-            missing_inputs = test_run.runnable.missing_inputs(params[:inputs])
-
+            check_runnable = repo.build_entity(create_params(params)).runnable
```

Comment:
Is there a reason not to just call this runnable? Also, this code isn't nil-safe.

Reply:
Nope. I was initially concerned about variable shadowing/overwriting, but that's an unfounded concern.

I renamed check_runnable and added some nil safety (I think I covered all the spots that needed checks, but let me know if I missed any)